src/test/kotlin/com/github/igorperikov/mightywatcher/external/RestGithubApiClientIntegrationTest.kt

```diff
@@ -1,15 +1,48 @@
 package com.github.igorperikov.mightywatcher.external
 
+import com.github.igorperikov.mightywatcher.Launcher
 import com.github.igorperikov.mightywatcher.utils.Iso8601Formatter
+import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.inject
 import java.time.LocalDateTime
 import java.time.Month
 import java.time.ZoneOffset
 
 private const val mightyWatcherGithubIntegrationTokenName = "MIGHTY_WATCHER_GITHUB_INTEGRATION_TOKEN"
 
-class RestGithubApiClientIntegrationTest {
+class RestGithubApiClientIntegrationTest : KoinTest {
+
+    private val githubApiClient by inject<GithubApiClient>()
+
+    private val testModule = module {
+        single { Launcher.initHttpClient(getProperty(mightyWatcherGithubIntegrationTokenName)) }
```

Comment:
why do we need any connection with Launcher here?

Reply:
I can move HttpClient initialization logic to GithubHttpClientFactory so it will have a connection with this factory.

Reply:
Furthermore, I think bringing DI framework here increases complexity (unnecessarily). It would be better, if the test would focus on testing url's, parameters, data formats and authorization only. We can have separate test for asserting that DI ecosystem builds client properly but not both at the same time. Do you agree with that? Do we really need to pass underlying http client via constructor parameter?

Reply:
Yes, we need to decouple RestGithubApiClient from specific HttpClient implementation, isn't it a point of dependency injection? When I did it I was thinking about the next issue #64, to test it I didn't want to send 5000 requests to Github and check if it degrades gracefully, what I did want is to mock HttpClient with a smaller x-rate-limit. I don't mind to avoid DI in tests and can remove it if you want me to, but in my opinion, RestGithubApiClient should be decoupled from HttpClient implementation.

Reply:
> what I did want is to mock HttpClient with a smaller x-rate-limit

you're right, totally makes sense

> I don't mind to avoid DI in tests and can remove it if you want me to, but in my opinion, RestGithubApiClient should be decoupled from HttpClient implementation.

is that doable to achieve both - remove DI from test and keep RestGithubApiClient decoupled from specific HttpClient implementation?

Reply:
Please see the following commit.

Reply:
> Please see the following commit.

Nice, I like this one 👍

Reply:
> When I did it I was thinking about the next issue #64

Let's discuss what you have in mind first? here - #64